src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -222,6 +225,106 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         Automation::AutomationProperties::SetFullDescription(UseDesktopImageCheckBox(), unbox_value<hstring>(backgroundImgCheckboxTooltip));
 
         Automation::AutomationProperties::SetName(DeleteButton(), RS_(L"Profile_DeleteButton/Text"));
+
+        if (!_FontList)
+        {
+            _UpdateFontList();
+        }
+    }
+
+    void Profiles::_UpdateFontList()
+    try
+    {
+        // initialize font list
+        _FontList = single_threaded_observable_vector<hstring>();
+
+        // get a DWriteFactory
+        ::IUnknown* unknownFactory;
+        THROW_IF_FAILED(DWriteCreateFactory(
+            DWRITE_FACTORY_TYPE_SHARED,
+            __uuidof(::IDWriteFactory),
+            &unknownFactory));
+
+        ::IDWriteFactory* dwriteFactory;
+        THROW_IF_FAILED(unknownFactory->QueryInterface(&dwriteFactory));
+
+        com_ptr<::IDWriteFactory> factory{};
+        factory.attach(dwriteFactory);
+
+        // get the font collection; subscribe to updates
+        com_ptr<::IDWriteFontCollection> fontCollection;
+        THROW_IF_FAILED(factory->GetSystemFontCollection(fontCollection.put(), TRUE));
```

Comment:
If this first bit fails, should we reset the _FontList to nullptr, so we try building it again?